InternxtDesktop/Views/Backups/BackupContentNavigator.swift

```diff
@@ -150,17 +151,75 @@ struct BackupContentNavigator: View {
     }
     
     private func downloadItem() {
-        let alert = NSAlert()
-        alert.messageText = "Download not implemented yet"
-        alert.informativeText = "You triggered a download, but this feature is not yet implemented"
         
-        alert.runModal()
+        
+        if(backupsService.backupDownloadStatus == .InProgress) {
+            let alert = NSAlert()
+            let title = NSLocalizedString("BACKUP_DOWNLOAD_IN_PROGRESS_ALERT_TITLE", comment: "")
+            let message = NSLocalizedString("BACKUP_DOWNLOAD_IN_PROGRESS_ALERT_MESSAGE", comment: "")
+            alert.messageText = title
+            alert.informativeText = message
+            alert.runModal()
+            return
+        }
+        
+        let panel = NSOpenPanel()
+        panel.allowsMultipleSelection = false
+        panel.canChooseDirectories = true
+        panel.canChooseFiles = false
+        let panelResponse = panel.runModal()
+        if(panelResponse == .OK) {
+            guard let url = panel.url else {
+                return
+            }
+            
+            
+            Task {
+                do {
+                    // download root folder
+                    if viewModel.navigationLevels.count == 1 && selectedFolderListItem == nil && selectedId == nil{
```

Comment:
this is hard to read, we can save this logic to a variable, which will improve the readability, an example:

`let currentLevelIsRoot = viewModel.navigationLevels.count == 1`

Then use the variable, it explains better the purpose